core/quality/sender.go

```diff
@@ -123,58 +141,63 @@ func (sender *Sender) Subscribe(bus eventbus.Subscriber) error {
 
 // sendSessionData sends transferred information about session.
 func (sender *Sender) sendSessionData(e connection.AppEventConnectionStatistics) {
-	if !e.SessionInfo.IsActive() {
+	if e.SessionInfo.SessionID == "" {
 		return
 	}
-	currentSession := e.SessionInfo
+
 	sender.sendEvent(sessionDataName, sessionDataContext{
-		Rx: e.Stats.BytesReceived,
-		Tx: e.Stats.BytesSent,
-		sessionContext: sessionContext{
-			ID:              string(currentSession.SessionID),
-			Consumer:        currentSession.ConsumerID.Address,
-			Provider:        currentSession.Proposal.ProviderID,
-			ServiceType:     currentSession.Proposal.ServiceType,
-			ProviderCountry: currentSession.Proposal.ServiceDefinition.GetLocation().Country,
-			ConsumerCountry: sender.originCountry(),
-		},
+		Rx:             e.Stats.BytesReceived,
+		Tx:             e.Stats.BytesSent,
+		sessionContext: sender.toSessionContext(e.SessionInfo),
+	})
+}
+
+func (sender *Sender) sendSessionEarning(e pingpongEvent.AppEventInvoicePaid) {
+	session, err := sender.recoverSessionContext(e.SessionID)
+	if err != nil {
+		log.Warn().Err(err).Msg("Can't recover session context")
+		return
+	}
+
+	sender.sendEvent(sessionTokensName, sessionTokensContext{
+		Tokens:         e.Invoice.AgreementTotal,
+		sessionContext: session,
 	})
 }
 
 // sendConnStateEvent sends session update events.
 func (sender *Sender) sendConnStateEvent(e connection.AppEventConnectionState) {
-	if !e.SessionInfo.IsActive() {
+	if e.SessionInfo.SessionID == "" {
```

Comment:
It was more clear before:
```suggestion
	if !e.SessionInfo.IsActive() {
```